build-system/tasks/dep-check.js

```diff
@@ -172,46 +171,61 @@ async function getEntryPointModule() {
     .map((file) => `import '../${file}';`)
     .join('\n');
   await fs.promises.writeFile(entryPointModule, entryPointData);
-  log('Added all entry points to', cyan(entryPointModule));
+  logLocalDev('Added all entry points to', cyan(entryPointModule));
   return entryPointModule;
 }
 
 /**
  * @param {string} entryPointModule
- * @return {!Promise<!ModuleDef>}
+ * @return {!ModuleDef}
  */
-function getModuleGraph(entryPointModule) {
-  let resolve;
-  const promise = new Promise((r) => {
-    resolve = r;
+async function getModuleGraph(entryPointModule) {
+  const esbuildBabelPlugin = {
```

Comment:
note: ideally we can extract the common bits of the esbuild babel plugin into a reused module. happy for this to be a future PR